File: EmergencyBurial/DataModel/Entities/Manage/ListItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DataModel.Entities;

public class ListItem
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.None)]
    [Required]
    public int Key { get; set; }
    
    public int? ListTypeId { get; set; }

    [Required]
    [MaxLength(150)]
    public string Text { get; set; }

    [MaxLength(250)]
    public string Description { get; set; }
    
    [MaxLength(10)]
    public int? ListItemDepId { get; set; }

    [MaxLength(250)]
    public string value { get; set; }

    public short Order { get; set; }
    
    [ForeignKey(nameof(ListTypeId))]
    public virtual ListType ListType { get; set; }
    
    [ForeignKey(nameof(ListItemDepId))]
    public virtual ListItem ListItemDep { get; set; }
}

File: WebApp/src/app/shared/model/list-item.ts
export interface IListItem {
  Id?: number;
  Key?: number;
  Text?: string;
  ListTypeId?: number;
  ListItemDepId?: number;
  IsValid?: boolean;
  Description?: string;
}

export interface IOptionItem {
  Value?: number;
  Text?: string;
}

export class ListItem implements IListItem {
  Text?: string;
  Description?: string;
  constructor(public Key?: number, public ListTypeId?: number,public IsValid: boolean = true) {
  }
}


